### ISSUE TITLE
Recent update seems to have broken this.

### ISSUE THREAD

**Issue by dreamwraith** [aaron-yang0327-development/ValheimMods#2]
After installing the latest update, anytime someone pings on the map, while you have the map open, the map breaks, and the pin icons separate from the map layer for a period of time.

Also, the mod generates MASSIVE lag when opening the map, and the console is spammed with critical red error messages, usually saying that the object id (0, 0, 0) already exists.

The map becomes slow to load and then will load without pins, etc.

Something about the update has become broken.

**Comment by Pkrakmo** [comment]
Will also break 'Visible to other players' when playing on a dedicated server

**Comment by aaron-yang0327-development** [comment]
will take a loot at this, I can't test out multiplayer stuffs so I can only work on it through assumptions on what would happen if player B does something and player A observes.
Can you send the log for a time where the error happens?

**Comment by aaron-yang0327-development** [comment]
thank you very much for the insights, I'll try to figure out what might cause this, on first look it seems to be some incompatibilities with server side map, but upon testing on a single player world, it persists even if PinAssistant is the only one existing. 

I found out I removed a check for when a ping is being added which led to errors, I thought it wasn't necessary until I read it again. Since the error happens the pinging completely stops which the game finds out that the addition of the ping didn't push through, which then tries to re add it and a loop of constant ping adding happens which freezes the map.

It also happens when you yourself joins a single player world as there's a shout ping whenever you join, but it wasn't noticeable as the ping was invisible cause of my errors I assume. middle clicking to ping as well also does this.

I'll send a hotfix for this in a few minutes and another update for better handling on special pins as my old code was "game adds location, pings, spawn points, etc." and then I remove it manually, when instead I could have just ignored this ping entirely instead through a different approach.

tl;dr; not a mod incompatibility with other mods, it's an overlook that I accidentally made when trying to clean up my code. Will send a fix soon in a few minutes and another update somewhere under 24 hours to code a better system to handling ignoring of special pins.

unsure about @Pkrakmo issues if it coincides with the freezing, but it should be fixed now, I'm not sure if the other problems you mentioned like the lag and loading without pins is related to the fix but let me know if there's still problems!

**Comment by dreamwraith** [comment]
Sounds like you found the right of it. Happy to test when I'm able to for anything multiplayer while we are doing our current MP run 👍

I'll capture logs next time, too

**Comment by aaron-yang0327-development** [comment]
new update released over at thunderstore and nexus, I'll leave this issue open so you guys can let me know if there's no more problems.

**Comment by dreamwraith** [comment]
Thanks Aaron, fix works great. Really appreciate your quick efforts on this!-

**Comment by aaron-yang0327-development** [comment]
no problem, hope you enjoy the mod
